flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/ZooKeeperLeaderElectionConnectionHandlingTest.java

```diff
@@ -131,7 +130,8 @@ private void runTestWithZooKeeperConnectionProblem(
             Problem problem)
             throws Exception {
         CuratorFramework client =
-                ZooKeeperUtils.startCuratorFramework(configuration, NoOpFatalErrorHandler.INSTANCE);
+                ZooKeeperUtils.startCuratorFramework(
+                        configuration, fatalErrorHandlerResource.getFatalErrorHandler());
```

Comment:
+1 for this. 
BTW,  I found that after removing the unhandled error listener in `LeaderElectionDriver` and `LeaderRetrievalDriver`. Some behavior of client created by using `NoOpFatalErrorHandler.INSTANCE` in test may be mute (without triggering fatal error setting in `TestingContender`). I will look over this again

Reply:
Good point, I think I should go over all the `startCuratorFramework(Configuration, FatalErrorHandler)` call sites and check that we don't only use a `NoOpFatalErrorHandler` if we expect an exception to be thrown.

Reply:
OK, I will do you a favor to double check .

Reply:
After going over all the call sites from `startCuratorFramework(Configuration, FatalErrorHandler)`.  I think only the place which use client to create `LeaderRetrieval/LeaderElectionService` should replace using `NoOpFatalErrorHandler` with the explicit `fatalErrorHandlerResource`

```
ZooKeeperLeaderRetrievalTest
ZooKeeperLeaderRetrievalConnectionHandlingTest
ZooKeeperLeaderElectionTest
ZooKeeperLeaderElectionConnectionHandlingTest
ZooKeeperHaServicesTest
ZooKeeperDefaultDispatcherRunnerTest
LeaderElectionTest
```

WDYT @tillrohrmann